examples/demo/src/themes/synthwaveTheme.ts

```diff
@@ -0,0 +1,263 @@
+        },
+    },
+});
+
+const Input = () => ({
+    MuiOutlinedInput: {
+        styleOverrides: {
+            input: {
+                padding: 16,
+            },
+        },
+    },
+});
+
+const ItemLink = (theme: Theme) => ({
+    RaMenuItemLink: {
+        styleOverrides: {
+            root: {
+                borderLeft: `3px solid ${theme.palette.primary.contrastText}`,
+                '&:hover': {
+                    borderRadius: '0px 100px 100px 0px',
+                },
+                '&.RaMenuItemLink-active': {
+                    borderLeft: `3px solid ${theme.palette.primary.main}`,
+                    borderRadius: '0px 100px 100px 0px',
+                    backgroundImage: `linear-gradient(98deg, ${theme.palette.primary.light}, ${theme.palette.primary.dark} 94%)`,
+                    boxShadow: theme.shadows[1],
+                    color: theme.palette.primary.contrastText,
+
+                    '& .MuiSvgIcon-root': {
+                        fill: theme.palette.primary.contrastText,
+                    },
+                },
+            },
+        },
+    },
+});
+
+const Layout = () => ({
+    RaLayout: {
+        styleOverrides: {
+            root: {
+                marginTop: 20,
+            },
+        },
+    },
+});
+
+const Paper = (theme: Theme) => {
+    const shadows = [
+        alpha(theme.palette.primary.main, 0.2),
+        alpha(theme.palette.primary.main, 0.1),
+        alpha(theme.palette.primary.main, 0.05),
+    ];
+    return {
+        MuiPaper: {
+            styleOverrides: {
+                elevation1: {
+                    boxShadow: `${shadows[0]} -2px 2px, ${shadows[1]} -4px 4px,${shadows[2]} -6px 6px`,
+                },
+                root: {
+                    backgroundClip: 'padding-box',
+                },
+            },
+        },
+    };
+};
+
+const Table = () => ({
+    MuiTableRow: {
+        styleOverrides: {
+            root: {
+                '&:last-child td': { border: 0 },
+            },
+        },
+    },
+    MuiTableCell: {
+        styleOverrides: {
+            root: {
+                padding: 16,
+            },
+        },
+    },
+});
+
+const TextField = () => ({
+    MuiTextField: {
+        defaultProps: {
+            variant: 'outlined' as const,
+        },
+    },
+});
+
+const Toolbar = (theme: Theme) => ({
+    MuiToolbar: {
+        styleOverrides: {
+            root: {
+                backgroundColor: theme.palette.background.default,
+                color: theme.palette.text.primary,
+            },
+            regular: {
+                backgroundColor: theme.palette.background.paper,
+            },
+        },
+    },
+});
+
+const componentsOverrides = (theme: Theme) => ({
+    ...AppBar(theme),
+    ...Button(),
+    ...Datagrid(theme),
+    ...FormControl(),
+    ...Input(),
+    ...ItemLink(theme),
+    ...Layout(),
+    ...Paper(theme),
+    ...Table(),
+    ...TextField(),
+    ...Toolbar(theme),
```

Comment:
I'm not sure those independent functions bring much clarity. Can you merge their content here like in the other themes?